src/openzaak/utils/permissions.py

```diff
@@ -138,13 +145,67 @@ def has_object_permission(self, request: Request, view, obj) -> bool:
         if not self.permission_fields:
             return request.jwt_auth.has_auth(scopes_required, component)
 
-        main_resource = self.get_main_resource()
+        main_resource = self.get_main_resource(self.main_resource)
 
         if view.__class__ is main_resource:
             main_object = obj
         else:
             main_object = self.get_main_object(obj, view.permission_main_object)
 
-        main_object_data = self.format_data(main_object, request)
-        fields = self.get_fields(main_object_data)
+        main_object_data = self.format_data(main_object, request, main_resource)
+        fields = self.get_fields(main_object_data, self.permission_fields)
         return request.jwt_auth.has_auth(scopes_required, component, **fields)
+
+
+class MultipleObjectsAuthRequired(AuthRequired):
+    permission_fields: dict
+    main_resources: dict
+
+    def get_field_viewset(self, request, viewset, action):
+        field_viewset = import_string(viewset)()
+        field_viewset.request = request
+        field_viewset.format_kwarg = "json"
+        field_viewset.action = action
+
+        return field_viewset
+
+    @convert_cmis_adapter_exceptions
+    def has_permission(self, request, view):
+        self.has_handler(request, view)
+
+        if bypass_permissions(request):
+            return True
+
+        if view.action is None and isinstance(view, ViewSetMixin):
+            return True
+
+        if not view.viewset_classes:
+            return False  # TODO could validate scopes on RegisterDocumentViewSet but component should be defined somewhere or raise ImproperlyConfigured
+
+        for field, viewset in view.viewset_classes.items():
+            # CatalogusAutorisatie is cached in JWTAuth
+            if hasattr(request.jwt_auth, "_autorisaties"):
+                del request.jwt_auth._autorisaties
```

Comment:
Why is this needed?

Reply:
the catalogusAuthorisaties are cached in _autorisaties which and filtered by component, so it gets all cat autorisaties for documenten which are then also incorrectly used for zaken